orbit/pkg/platform/platform_windows.go

```diff
@@ -341,3 +344,203 @@ func GetSMBiosUUID() (string, UUIDSource, error) {
+		// by first renaming current orbit file and then recreating recreating the symlink
+
+		// renaming locked binary to a different file, the process will keep running, but it will be renamed
+		// target process is not terminated on purpose to avoid potential erros
+		targetOrbitPath := orbitPath + "." + strings.ToUpper(uuid.New().String())
+		if err := os.Rename(orbitPath, targetOrbitPath); err != nil {
+			return fmt.Errorf("rename: %w", err)
+		}
+
+		// regenerating the symlink
+		if err := os.Symlink(targetSymlinkPath, orbitPath); err != nil {
+			return fmt.Errorf("symlink current: %w", err)
+		}
+
+		// the renamed binary file is locked because is used by a running process
+		// so only thing possible is to mark it to be deleted upon reboot by using MOVEFILE_DELAY_UNTIL_REBOOT flag
+		// https://learn.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-movefileexw
+		if err := windows.MoveFileEx(windows.StringToUTF16Ptr(targetOrbitPath), nil, windows.MOVEFILE_DELAY_UNTIL_REBOOT); err != nil {
+			return fmt.Errorf("movefileex: %w", err)
+		}
+	}
+
+	return nil
+}
+
+// isRunningAsSystem checks if the current process is running as SYSTEM
+func isRunningAsSystem() (bool, error) {
+	// getting the current process token
+	token, err := windows.OpenCurrentProcessToken()
+	if err != nil {
+		return false, err
+	}
+	defer token.Close()
+
+	// getting the current process user
+	user, err := token.GetTokenUser()
+	if err != nil {
+		return false, err
+	}
+
+	// checking if the current process user is SYSTEM
+	if windows.EqualSid(user.User.Sid, constant.SystemSID) {
+		return true, nil
+	}
+
+	return false, nil
+}
+
+// isRunningFromStagingDir checks if the current process is running from the staging directory
+func isRunningFromStagingDir() (bool, error) {
+	// getting current working directory
+	execPath, err := getWorkingPath()
+	if err != nil {
+		return false, err
+	}
+
+	// checking if the current executable directory is the staging directory
+	if !strings.HasSuffix(strings.ToLower(execPath), "staging") {
+		return false, errors.New("not running from the staging directory")
+	}
+
+	return true, nil
+}
+
+// shouldQuirksRun determines if the software update quirks should be run
+// by checking if process is running as system and from staging directory
+// we can relax the constrains a bit if needed and just check for SYSTEM execution context
+func shouldQuirksRun() bool {
+	isSystem, err := isRunningAsSystem()
+	if err != nil {
+		return false
+	}
+
+	isStagingDir, err := isRunningFromStagingDir()
+	if err != nil {
+		return false
+	}
+
+	if isSystem && isStagingDir {
+		return true
+	}
+
+	return false
+}
+
+// RunUpdateQuirks runs the best-effort software update quirks
+// There is no logging support in this function, as it is called before the logging system is initialized
+func RunUpdateQuirks() {
+	if shouldQuirksRun() {
+		// Fixing the symlink not present quirk
+		fixSymlinkNotPresent()
```

Comment:
Let's add a comment that this is a best effort fix, any error in `fixSymlinkNotPresent` is ignored.

Reply:
just added the comment. Good point. The nature of the fix is to not interrupt the update in any way, so good to clarify that.